cmd/snap-bootstrap/README.md

```diff
@@ -12,10 +14,64 @@ _snap-bootstrap_ is the main executable that is run during the initramfs stage o
+11. If we are either not trying a recovery system or we are in a consistent state and are trying a recovery system, then we enter the following magical state machine. This state diagram essentially allows recover mode to be extra robust against failure modes like having a partition disappear, some keys not being able to unlock some of the partitions, etc. and is referred to as "degraded mode", specifically if we don't use all the happy paths then we are in a "degraded" recover mode as opposed to being in a normal recover mode.
 
-When booting into recover mode, _snap-bootstrap_ has some additional logic setup to try and be as robust as possible. This logic is fairly complicated and best explained in the following state diagram showing the states and transitions that _snap-bootstrap_ operates in during recover mode, which has been called degraded mode.
 
 ![](/cmd/snap-bootstrap/degraded-recover-mode.svg)
 
-The above state diagram was made with https://app.diagrams.net/ and can be imported by opening the SVG file in this directory there.
\ No newline at end of file
+
+The above state diagram was made with https://app.diagrams.net/ and can be imported by opening the SVG file in this directory there.
+
+12. After exiting the state machine (in all cases), we will then again consider if we are trying a recovery system, and if we are we will inspect if the state machine degraded at all (meaning that the "happy path" for unlocking disks and mounting partitions was not fully executed and we had to use an alternative option at least one time), and if the state machine outputs a degraded state, then we mark the recovery system as a failure and go back to run mode. If it was successful, we mark it as successful and then reboot to run mode as well.
+13. In the case where we were not trying a recovery system and we were in fact just booting into the recovery system for normal recover mode, we will first write out a file `degraded.json` with information about whether or not the state machine output us in a degraded state or not. This may affect some choices userspace makes when we get there.
+14. If the state machine exited in a state that was at least sufficiently usable such that we can trust the data partition unlocked and mounted, we will then copy some files from the data partition to our tmpfs root filesystem, such as authentication files like ssh keys, networking configuration, and other miscellaneous files like the clock sync file in case the system does not have a battery powered RTC. If we didn't trust the data partition, then "safe" defaults will be used instead. This is to prevent a situation wherein we don't "trust" the data partition enough (but perhaps we did trust ubuntu-save when unlocking it) to copy authentication files over, but then we leave console-conf in such a state where it could allow an attacker to create their own new account and then exfiltrate secret data from the trusted ubuntu-save.
```

Comment:
```suggestion
14. If the state machine exited in a state that was at least sufficiently usable, such that we can trust the data partition unlocked and mounted, we will then copy some files from the data partition to our tmpfs root filesystem. These could include authentication files, such as ssh keys, networking configuration, and other miscellaneous files like the clock sync file for systems without a battery powered RTC. If we didn't trust the data partition, then "safe" defaults will be used instead. This is to prevent a situation wherein we don't "trust" the data partition enough (but perhaps we did trust ubuntu-save when unlocking it) to copy authentication files over, but then we leave console-conf in such a state where it could allow an attacker to create their own new account and then exfiltrate secret data from the trusted ubuntu-save.
```